plugins/modules/aci_interface_policy_storm_control.py

```diff
@@ -0,0 +1,434 @@
+        name_alias=dict(type="str"),
+        storm_control_types=dict(type="str", choices=["all_types", "unicast_broadcast_multicast"]),
+        all_types_configuration=dict(type="dict", options=storm_control_policy_rate_spec(), aliases=["all_types"]),
+        broadcast_configuration=dict(type="dict", options=storm_control_policy_rate_spec(), aliases=["broadcast"]),
+        multicast_configuration=dict(type="dict", options=storm_control_policy_rate_spec(), aliases=["multicast"]),
+        unicast_configuration=dict(type="dict", options=storm_control_policy_rate_spec(), aliases=["unicast"]),
+        storm_control_action=dict(type="str", choices=["drop", "shutdown"]),
+        storm_control_soak_action=dict(type="int"),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["storm_control_policy"]],
+            ["state", "present", ["storm_control_policy"]],
+        ],
+        mutually_exclusive=[
+            ("all_types_configuration", "broadcast_configuration"),
+            ("all_types_configuration", "multicast_configuration"),
+            ("all_types_configuration", "unicast_configuration"),
+        ],
+    )
+
+    aci = ACIModule(module)
+
+    storm_control_policy = module.params.get("storm_control_policy")
+    description = module.params.get("description")
+    state = module.params.get("state")
+    name_alias = module.params.get("name_alias")
+    storm_control_types = MATCH_STORM_CONTROL_POLICY_TYPE_MAPPING.get(module.params.get("storm_control_types"))
+    storm_control_action = module.params.get("storm_control_action")
+    storm_control_soak_action = module.params.get("storm_control_soak_action")
+
+    rates_input = {}
+
+    stom_control_types_configs = [
+        dict(
+            config_input=module.params.get("all_types_configuration"),
+            rates=dict(rate=dict(percentage="rate", pps="ratePps"), burst_rate=dict(percentage="burstRate", pps="burstPps")),
+        ),
+        dict(
+            config_input=module.params.get("broadcast_configuration"),
+            rates=dict(rate=dict(percentage="bcRate", pps="bcRatePps"), burst_rate=dict(percentage="bcBurstRate", pps="bcBurstPps")),
+        ),
+        dict(
+            config_input=module.params.get("multicast_configuration"),
+            rates=dict(rate=dict(percentage="mcRate", pps="mcRatePps"), burst_rate=dict(percentage="mcBurstRate", pps="mcBurstPps")),
+        ),
+        dict(
+            config_input=module.params.get("unicast_configuration"),
+            rates=dict(rate=dict(percentage="uucRate", pps="uucRatePps"), burst_rate=dict(percentage="uucBurstRate", pps="uucBurstPps")),
+        ),
+    ]
+
+    for config in stom_control_types_configs:
+        config_input = config.get("config_input")
+        rates = config.get("rates")
+        if config_input is not None:
+            if config_input.get("rate_type") == "percentage":
```

Comment:
What happens if rate_type is not provided? This will be None so both conditionals will be skipped? Should we restrict the input of storm_control_policy_rate_spec to have certain input as required since it seems we are depending on it in conditionals?